src/libopensc/sc.c

```diff
@@ -897,40 +896,22 @@ int _sc_parse_atr(sc_reader_t *reader)
 	return SC_SUCCESS;
 }
 
-static void init_page_size()
-{
-	if (page_size == 0) {
-#ifdef _WIN32
-		SYSTEM_INFO system_info;
-		GetSystemInfo(&system_info);
-		page_size = system_info.dwPageSize;
-#else
-		page_size = sysconf(_SC_PAGESIZE);
-		if ((long) page_size < 0) {
-			page_size = 0;
-		}
-#endif
-	}
-}
-
 void *sc_mem_secure_alloc(size_t len)
 {
 	void *p;
 
-	init_page_size();
-	if (page_size > 0) {
-		size_t pages = (len + page_size - 1) / page_size;
-		len = pages * page_size;
-	}
-
-	p = calloc(1, len);
-	if (p == NULL) {
-		return NULL;
-	}
 #ifdef _WIN32
-	VirtualLock(p, len);
+	p = VirtualAlloc(NULL, len, MEM_COMMIT, PAGE_READWRITE);
```

Comment:
The [example code from MSDN](https://learn.microsoft.com/en-us/windows/win32/Memory/memory-protection-constants) still uses the pagesize from sysinfo for the allocation. Aren't you  missing that part in your code?

Reply:
you're right, sorry for the noise and thanks for the clarification